truss/templates/shared/logging.py

```diff
@@ -22,11 +22,37 @@ def filter(self, record: logging.LogRecord) -> bool:
         )
 
 
+class StreamToLogger:
+    """
+    StreamToLogger redirects stdout and stderr to logger
+    """
+
+    def __init__(self, logger, log_level, stream):
+        self.logger = logger
+        self.log_level = log_level
+        self.stream = stream
+
+    def __getattr__(self, name):
+        # we need to pass `isatty` from the stream for uvicorn
+        # this is a more general, less hacky fix
+        return getattr(self.stream, name)
+
+    def write(self, buf):
+        for line in buf.rstrip().splitlines():
+            self.logger.log(self.log_level, line.rstrip())
+
+    def flush(self):
+        pass
+
+
 def setup_logging() -> None:
     loggers = [logging.getLogger()] + [
         logging.getLogger(name) for name in logging.root.manager.loggerDict
     ]
 
+    sys.stdout = StreamToLogger(logging.getLogger(), logging.INFO, sys.stdout)  # type: ignore
```

Comment:
worth nothing that the original streams can be recovered from `sys._stdout` etc. It might actually be cleaner to rewrite this as 
```
 sys.stdout = StreamToLogger(logging.getLogger(), logging.INFO, sys._stdout) 
```